Refine cleanup logic in main.c

ChinYikMing opened the pull request.

jserv reviewed (changes requested):
Make both git commit message and the subject informative.

jserv commented on src/main.c:
```diff
@@ -224,6 +224,10 @@ int main(int argc, char **args)
+end:
     free(attr.data.user);
-    return 0;
+    if (prof_out_file)
```
Use `free(prof_out_file)` without `if` statement.

ChinYikMing replied:
> Use `free(prof_out_file)` without `if` statement.

Sure. As man page of `free` says: "If ptr is a NULL pointer, no operation is performed.".

ChinYikMing committed:
Refine cleanup logic in main.c

There are redundant codes for cleaning up the memory, thus refine them.
Also, prof_out_file should be freed in commit 0b8a712 but it did not,
so it does in this commit.
